### PR TITLE
Updated Paper + Fix Result Set Handling

### PR DESCRIPTION
**Pull request by darbyjack**
The latest build of Paper introduced an update to the mysql-connector. One of the changes made it more strict on the usage of `rs.first()`. Somewhere in Prism is logic to handle going forward and backwards, but with `rs.first()`, you can not do that or it will produce the following error: https://paste.gg/p/anonymous/9010ae83f15249d281d1917059c086aa

This PR changes the usage of `rs.first()` to `rs.next()` which is the "proper" usage of handling result sets.

### CODE REVIEW

**Review by Ofacy — APPROVED** [review]

**Review by Ofacy — APPROVED** [review]

**Comment by AzCreep** [comment]
I'm having that same error on my server how do i proced to fix it?

**Comment by TNTUP** [comment]
Having this issue aswell, any ETA to see this merged?

**Comment by NeverGrowUp** [comment]
Narimm/Add. Added these commits and confirmed fixes issue locally. Copy also located on Staff and working.

File: Prism/pom.xml [hunk 1]
```diff
@@ -34,7 +34,7 @@
         <dependency>
             <groupId>net.kyori</groupId>
             <artifactId>adventure-api</artifactId>
-            <version>4.1.1</version>
+            <version>4.7.0</version>
```
Comment (addstar) [inline Prism/pom.xml:37]: @darbyjack Just curious, is this version bump related to the fix? I assume not, but it wasn't mentioned in the PR description. It's not a blocker, but I want to be sure unrelated changes aren't being pulled in :slightly_smiling_face:

Comment (addstar) [reply Prism/pom.xml:37]: Also sorry we haven't been responsive lately, life getting in the way and all... I'm happy to approve this, just wanted to clarify regarding the `adventure` changes first

**Comment by addstar** [comment]
> Added these commits and confirmed fixes issue locally. Copy also located on Staff and working.

:clap: Also, thank you for testing this! :grin: I never get time for this kind of work any more, it makes it so much easier to approve when I know it's been tested/validated! :100:

File: Prism/pom.xml (hunk 1 above)
Comment (darbyjack) [reply Prism/pom.xml:37]: I want to say there was some error at some point related to it, yes. I think it was with the LegacySerializer / Deserializer. I don't have logs all the way from back then anymore but I made the change due to some error when I was trying to fix the MySQL.

Comment (addstar) [reply Prism/pom.xml:37]: All good, I think it's safe to merge this now :+1: Nice work, thanks!! :clap:

**Review by addstar — APPROVED** [review]

**Comment by addstar** [comment]
Having trouble getting all status checks to pass, but that's a problem with the tests, not your change... I'm working on it.

**Comment by addstar** [comment]
Wasn't able to get the Codacy status check to work (even though it's green) so I merged anyway. Thanks again for the contribution! :clap:

**Comment by AzCreep** [comment]
does this mean it's fixed or not yet?

**Comment by addstar** [comment]
> does this mean it's fixed or not yet?

The fix has been merged into the master code branch, but there is no build yet with the fix.

As per the repo [README](https://github.com/AddstarMC/Prism-Bukkit/blob/master/README.md#build-server), dev builds will be available [here](https://jenkins.addstar.com.au/job/Prism-Bukkit/lastSuccessfulBuild/). When it's built :slightly_smiling_face:
